wps_shared/wps_shared/db/crud/model_run_repository.py

```diff
@@ -70,3 +74,58 @@ def get_model_run_prediction(self, prediction_run: PredictionModelRunTimestamp,
             .first()
         )
         return prediction
+    
+    def get_prediction_model_run_timestamp_records(self, model_type: ModelEnum, complete: bool = True, interpolated: bool = True) -> List[Tuple[PredictionModelRunTimestamp, PredictionModel]]:
+        """Get prediction model run timestamps (filter on complete and interpolated if provided.)"""
+        query = (
+            self.session.query(PredictionModelRunTimestamp, PredictionModel)
+            .join(PredictionModelRunTimestamp, PredictionModelRunTimestamp.prediction_model_id == PredictionModel.id)
+            .filter(PredictionModel.abbreviation == model_type.value)
+        )
+        if interpolated is not None:
```

Comment:
Since defaults are being provided for `interpolated` and `completed` will these `if` statements ever be `True`?

Reply:
Hmm good point, this is existing code moved into a function here, but I think they should be boolean conditionals, not null checks.

Reply:
I cleaned this up and added tests around it in: https://github.com/bcgov/wps/pull/4469/commits/985cc40d8b663a6ee772eec738e7c5342d00bdb9